### PR TITLE
[feature-wip](new-scan)Add new file scan node

### PR DESCRIPTION
**Pull request by Jibing-Li**
# Proposed changes

Related pr: https://github.com/apache/doris/pull/11582
This is the new file scan node and scanner for external hms catalog.

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by SaintBacchus** [comment]
why add a New file scan node?  the file scan node is not public to our user, you can add your code in the file scan node

**Comment by morningman** [comment]
> why add a New file scan node? the file scan node is not public to our user, you can add your code in the file scan node

We are refactoring the whole scan node framework. all scan node will be derived from `VScanNode`.
And this is a bypass, after all is done, the old scan node will be removed.

File: be/src/vec/exec/scan/vscanner.h [hunk 1]
```diff
@@ -151,6 +154,11 @@ class VScanner {
     // watch to count the time wait for scanner thread
     MonotonicStopWatch _watch;
     int64_t _scanner_wait_worker_timer = 0;
+    int64_t _raw_rows_read = 0;
+    int64_t _num_rows_return = 0;
```
Comment (morningman) [inline be/src/vec/exec/scan/vscanner.h:158]: Not used?

File: be/src/vec/exec/scan/vscanner.h [hunk 2]
```diff
@@ -151,6 +154,11 @@ class VScanner {
     // watch to count the time wait for scanner thread
     MonotonicStopWatch _watch;
     int64_t _scanner_wait_worker_timer = 0;
+    int64_t _raw_rows_read = 0;
```
Comment (morningman) [inline be/src/vec/exec/scan/vscanner.h:157]: Not used?

File: be/src/vec/exec/scan/vscan_node.h [hunk 3]
```diff
@@ -34,7 +34,9 @@ class VSlotRef;
 class VScanNode : public ExecNode {
 public:
     VScanNode(ObjectPool* pool, const TPlanNode& tnode, const DescriptorTbl& descs)
-            : ExecNode(pool, tnode, descs), _runtime_filter_descs(tnode.runtime_filters) {}
+            : ExecNode(pool, tnode, descs), _runtime_filter_descs(tnode.runtime_filters) {
+        _runtime_filter_descs = tnode.runtime_filters;
```
Comment (morningman) [inline be/src/vec/exec/scan/vscan_node.h:38]: Duplicated?

Comment (Jibing-Li) [reply be/src/vec/exec/scan/vscan_node.h:38]: Double checked, it is duplicated. This line ```_runtime_filter_descs(tnode.runtime_filters)``` was added in your new pr, will remove the duplicated line.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**